pkg/reconciler/instances/rma/lazy_client.go

```diff
@@ -25,7 +25,7 @@ type LazyKubeClient struct {
 
 func (c *LazyKubeClient) init() error {
 	c.initClient.Do(func() {
-		c.client, c.clientErr = kubeclient.NewInClusterClient(c.log)
+		c.client, c.clientErr = reconK8s.NewInClusterClientSet(c.log)
```

Comment:
Wouldn't it be better to initialize only a new `InClusterClient` here, and return a new `ClientSet` from reconciler kubeclient at every call to LazyKubeClient.ClientSet() instead of returning the same clientset instance?

In this case different workers would use different K8S client with separate client-side throttling (rate-limiting) bucket, HTTP client, etc. instead of a static, "global" shared clientset

Reply:
Ideally I wanted to do it like that, but the in-cluster client generally has less things than the local one (e.g. no kubeconfig). So it would break the `Client` interface that we have.

It is a good improvement I would like to add but I already am now constantly having to fix issues because of teams misusing the package so it would be good to at least get this merged ASAP and then improve upon it.